openedx/core/djangoapps/schedules/management/commands/tests/send_email_base.py

```diff
@@ -169,13 +175,27 @@ def test_schedule_bin(self, schedule_count, mock_metric, mock_ace):
             self.assertEqual(mock_schedule_send.apply_async.call_count, schedule_count)
             self.assertFalse(mock_ace.send.called)
 
+    def _schedule_factory(self, offset=None, **factory_kwargs):
+        _, _, target_day, upgrade_deadline = self._get_dates(offset=offset)
+        factory_kwargs.setdefault('start', target_day)
+        factory_kwargs.setdefault('upgrade_deadline', upgrade_deadline)
+        factory_kwargs.setdefault('enrollment__course__self_paced', True)
+        if hasattr(self, 'experience_type'):
+            factory_kwargs.setdefault('experience__experience_type', self.experience_type)
+        schedule = ScheduleFactory(**factory_kwargs)
+        course_id = schedule.enrollment.course.id
+        if course_id not in self._verified_modes:
+            CourseModeFactory(
+                course_id=schedule.enrollment.course.id,
+                mode_slug=CourseMode.VERIFIED,
+                expiration_datetime=datetime.datetime.now(pytz.UTC) + datetime.timedelta(days=30),
+            )
+            self._verified_modes.add(course_id)
```

Comment:
Can we rename `_verified_modes` to something like `_courses_with_verified_modes` since it's a cache of course-keys rather than the modes themselves.

Reply:
Done!